scripts/linters/pylint_extensions.py

```diff
@@ -1858,6 +1858,37 @@ def process_tokens(self, tokens):
                         line=line_num)
 
 
+class InequalityWithNoneChecker(checkers.BaseChecker):
+    """Custom pylint checker prohibiting use of "if x != None" and
+    enforcing use of "if x is not None" instead.
+    """
+
+    __implements__ = interfaces.IAstroidChecker
+
+    name = 'inequality-with-none'
+    priority = -1
+    msgs = {
+        'C0030': (
+            'Please refrain from using "x != None" '
+            'and use "x is not None" instead.',
+            'inequality-with-none',
+            'Use "is" to assert equality or inequality against None.'
+        )
+    }
+
+    def visit_compare(self, node):
+        """Called for comparisons (a != b).
+
+        Args:
+            node: astroid.node.compare. Node comparing two values.
+        """
+
+        ops = node.ops
+        for (operator, operand) in ops:
+            if operator == '!=' and operand.as_string() == 'None':
```

Comment:
should we use isInstance(operand, node.Constant.NoneType) instead of as_string? (not sure!)

Also, can you confirm this check works for the following case (add a test case for this):

```
if something != (
    None):

    ....
```

Reply:
I can't find an Astroid type for NoneType, but only an umbrella type `astroid.Const` which is more than we want. I've also tested the above mentioned case manually by running the linter on a file I changed to contain that pattern, and it shows that using `as_string()` actually works since I guess it gives the **type** of the operand as a string.
![image](https://user-images.githubusercontent.com/16142055/97978166-0a424000-1d82-11eb-9ae8-4ca2ba5f254e.png)
![image](https://user-images.githubusercontent.com/16142055/97978179-11694e00-1d82-11eb-809a-94b3cf0c1f01.png)
Aside from that, I've added the test cases and will include them in the latest revision. Thanks!

Reply:
Sorry I wanted to say `node.Const.NoneType` instead of `node.Constant.NoneType`. (I found it [here](http://pylint.pycqa.org/projects/astroid/en/latest/api/astroid.nodes.html#astroid.nodes.Const))